c7n/resources/rds.py

```diff
@@ -274,6 +274,133 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('minor-upgrade-flag')
+class NoMinorUpgradeFlag(Filter):
+
+    schema = type_schema('minor-upgrade-flag', value={'type': 'boolean'})
+
+    def process(self, resources, event=None):
+        if self.data.get('value', True):
+            return [r for r in resources if not r['AutoMinorVersionUpgrade']]
+        else:
+            return [r for r in resources if r['AutoMinorVersionUpgrade']]
+
+
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+
+    schema = type_schema('upgrade-available')
+
+    def _list_versions(self, client, resource):
+        from distutils.version import StrictVersion as sv
+        versions = client.describe_db_engine_versions(
+            Engine=resource['Engine'],
```

Comment:
your fetching this info per database instance, when it only needs to be fetched once per engine.